linkerd/proxy/transport/src/connect.rs

```diff
@@ -38,12 +37,13 @@ impl<T: Param<Remote<ServerAddr>>> Service<T> for ConnectTcp {
             let io = TcpStream::connect(&addr).await?;
             super::set_nodelay_or_warn(&io);
             let io = super::set_keepalive_or_warn(io, keepalive)?;
+            let local_addr = io.local_addr()?;
             debug!(
-                local.addr = %io.local_addr().expect("cannot load local addr"),
+                local.addr = %local_addr,
```

Comment:
this will now bail rather than panicking if the local address is missing...is that what we want? should we be logging an error or something before bailing?

Reply:
This will manifest as a connection error and be logged where those errors are handled.